tools/restatectl/src/commands/node/disable_node_checker.rs

```diff
@@ -77,6 +107,69 @@ impl<'a, 'b> DisableNodeChecker<'a, 'b> {
         Ok(())
     }
 
+    /// Checks whether it is safe to disable the worker. A worker is safe to disable if it is not
+    /// used in any partition replica set, and it cannot be added to any future replica sets.
+    async fn safe_to_disable_worker(
+        &self,
+        node_id: PlainNodeId,
+        node_config: &NodeConfig,
+    ) -> Result<(), DisableWorkerError> {
+        match node_config.worker_config.worker_state {
+            WorkerState::Active => {
+                return Err(DisableWorkerError::Active);
+            }
+            WorkerState::Draining => {
+                // we need to check whether the worker is part of any replica sets
+
+                // Unfortunately, there is no guarantee that a draining worker won't be added to any
+                // future replica set because the corresponding NodesConfiguration might not have
+                // been sent to all nodes of the cluster :-( We might want to wait for a given
+                // duration after marking a worker as draining before we consider it safe to
+                // disable it.
+            }
+            WorkerState::Provisioning | WorkerState::Disabled => {
+                return Ok(());
+            }
+        }
+
+        let mut replica_membership = JoinSet::new();
+
+        // We assume that the partition table does contain all relevant partitions. This will break
+        // once we support dynamic partition table updates. Once this happens, we need to wait until
+        // the draining worker state has been propagated to all nodes.
+        for partition_id in self.partition_table.iter_ids().copied() {
+            replica_membership.spawn({
+                let metadata_client = self.metadata_client.clone();
+
+                async move {
+                    // todo replace with multi-get when available
+                    let epoch_metadata = metadata_client
```

Comment:
I'm a little anxious about doing this on restatectl's side. I understand that epoch-metadata is the source-of-truth, but would using the view from `PartitionReplicaSetStates` be a possible alternative? We can expose those replica-sets via datafusion (I'm already on it).

Reply:
It's a bit more work but wouldn't it be even better to put the remove node operation in the `ClusterCtrlSvc`? Then we can use from places like the BYOC deployment controller. The source of truth behind that can still be the DF but we also have access to locally-cached metadata on the server.

Reply:
@pcholakov that should be our long-term goal,  yes.